core/src/main/scala/org/bitcoins/core/crypto/TxSigComponent.scala

```diff
@@ -53,6 +53,20 @@ sealed abstract class BaseTxSigComponent extends TxSigComponent {
   override def sigVersion = SigVersionBase
 }
 
+sealed abstract class P2SHTxSigComponent extends BaseTxSigComponent {
+  override def scriptPubKey: P2SHScriptPubKey =
+    output.scriptPubKey.asInstanceOf[P2SHScriptPubKey]
+
+  override def scriptSignature: P2SHScriptSignature = {
+    val s = transaction.inputs(inputIndex.toInt).scriptSignature
+    require(s.isInstanceOf[P2SHScriptSignature],
+            "Must have P2SHScriptSignature for P2SH, got: " + s)
+    s.asInstanceOf[P2SHScriptSignature]
```

Comment:
This require should be top-level rather than inside this `def`, and also there should be a `require` for the `output.scriptPubKey`